### ISSUE TITLE
[a11y]: [Windows High Contrast Mode]: [ContentSwitcher] Icons are not visible and selected item has no borders

### ISSUE THREAD

**Issue by xrissot-ibm** [carbon-design-system/carbon#19140]
### Package

@carbon/react

### Browser

Chrome

### Operating System

Windows

### Package version

v1.79

### Automated testing tool and ruleset

IBM Accessibility tool - needs review section 

### Description

Icon should be visible, borders also and way to determine which entry is the selected one

### WCAG 2.1 Violation

Verify High contrat mode on windows

### Reproduction/example

Use the playground for the component and activate HCM

### Steps to reproduce

Use the playground for the component and activate HCM.

Selected item icon is not visible (in Icons only mode) + borders and background for the selected value;
![Image](https://github.com/user-attachments/assets/223aba21-aa38-463a-81de-ce4d18508809)

![Image](https://github.com/user-attachments/assets/cfc55dd6-4ce4-4db6-a1d3-d334c6770749)

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by mariyageorge01** [comment]
@heloiselui @tay1orjones Can you please assign this issue to me?
